app/views/courses/CourseDetailsView.coffee

```diff
@@ -25,125 +25,64 @@ module.exports = class CourseDetailsView extends RootView
   constructor: (options, @courseID, @courseInstanceID) ->
     super options
     @ownedClassrooms = new Classrooms()
-    @ownedClassrooms.fetchMine({data: {project: '_id'}})
-    @supermodel.trackCollection(@ownedClassrooms)
-    @courseID ?= options.courseID
-    @courseInstanceID ?= options.courseInstanceID
+    @courses = new Courses()
+    @course = new Course()
+    @levelSessions = new LevelSessions()
+    @courseInstance = new CourseInstance({_id: @courseInstanceID})
+    @owner = new User()
     @classroom = new Classroom()
-    @course = @supermodel.getModel(Course, @courseID) or new Course _id: @courseID
-    @listenTo @course, 'sync', @onCourseSync
-    if @course.loaded
-      @onCourseSync()
-    else
-      @supermodel.loadModel @course
-
-  getRenderData: ->
-    context = super()
-    context.campaign = @campaign
-    context.course = @course if @course?.loaded
-    context.courseInstance = @courseInstance if @courseInstance?.loaded
-    context.courseInstances = @courseInstances?.models ? []
-    context.levelConceptMap = @levelConceptMap ? {}
-    context.noCourseInstance = @noCourseInstance
-    context.noCourseInstanceSelected = @noCourseInstanceSelected
-    context.userLevelStateMap = @userLevelStateMap ? {}
-    context.promptForSchool = @courseComplete and not me.isAnonymous() and not me.get('schoolName') and not storage.load('no-school')
-    context
-
-  afterRender: ->
-    super()
-    if @supermodel.finished() and @courseComplete and me.isAnonymous() and @options.justBeatLevel
-      # TODO: Make an intermediate modal that tells them they've finished HoC and has some snazzy stuff for convincing players to sign up instead of just throwing up the bare CreateAccountModal
-      CreateAccountModal = require 'views/core/CreateAccountModal'
-      @openModalView new CreateAccountModal showSignupRationale: true
-
-  onCourseSync: ->
-    return if @destroyed
-    # console.log 'onCourseSync'
-    if me.isAnonymous() and (not me.get('hourOfCode') and not @course.get('hourOfCode'))
-      @noCourseInstance = true
-      @render()
-      return
-    return if @campaign?
-    campaignID = @course.get('campaignID')
-    @campaign = @supermodel.getModel(Campaign, campaignID) or new Campaign _id: campaignID
-    @listenTo @campaign, 'sync', @onCampaignSync
-    if @campaign.loaded
-      @onCampaignSync()
-    else
-      @supermodel.loadModel @campaign
-    @render()
+    @levels = new Levels()
+    @courseInstances = new CourseInstances()
+
+    @supermodel.trackRequest @ownedClassrooms.fetchMine({data: {project: '_id'}})
+    @supermodel.trackRequest(@courses.fetch().then(=>
+      @course = @courses.get(@courseID)
+    ))
+    sessionsLoaded = @supermodel.trackRequest(@levelSessions.fetchForCourseInstance(@courseInstanceID, {cache: false}))
+
+    @supermodel.trackRequest(@courseInstance.fetch().then(=>
+      return if @destroyed
```

Comment:
I'm ambivalent about them, but I'm leaning more towards being clearer about when the function ends, and having more parens in general.